assets/js/langRedirector.js

```diff
@@ -0,0 +1,45 @@
+// Wrap all this to not pollute global namespace
+{
+    // Generate an array of languages from page
+    // TODO: is there a way to do it static using Jekyll?
+    const JDM_LANGS = [...document.querySelectorAll('#language-dropdown > ul > li')].map(el => el.classList[0]);
+
+
+    function setLanguage(lang) {
+        path = lang;
+        // English is mounted on '/'
+        if (lang === 'en')
+            path = '';
+
+        location = location.origin+'/'+path;
+    }
+
+
+    function _tryLanguage(lang) {
+        if (JDM_LANGS.includes(lang))
+            setLanguage(lang);
+    }
+
+
+    function tryLanguage(lang) {
+        _tryLanguage(lang);
+    
+        // Test first two chars. Maybe we have not local, but more common language?
+        lang = lang.slice(0,2);
```

Comment:
Doesn't this actually extract the first three chars? How do you tell apart `zh-CN` and `zh-TW` if all you get is `zh-`?

Reply:
[`String#slice(int, int)`'s endIndex is exclusive.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice#Parameters)